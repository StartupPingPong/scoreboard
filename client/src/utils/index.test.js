import enzyme from "enzyme";
import { matchMaker, tournamentMaker, isPowerOfTwo } from "./";

describe("Given 2^n teams, it returns paired matches", () => {
  let teams = ["a", "b", "c", "d", "e", "f", "g", "h"];
  let incorrect = ["a", "b", "c"];
  let matches;
  let noMatches;

  beforeEach(() => {
    matches = matchMaker(teams);
    noMatches = matchMaker(incorrect);
  });

  it("does not create matches for number fo teams different than 2^n", () => {
    expect(noMatches.length).toEqual(0);
  });

  it("creates a matches array of half the length of teams", () => {
    expect(matches.length).toEqual(teams.length / 2);
  });
});

describe("Given an odd number of teams, it returns the first leg of a tournament", () => {
  let teams = ["a", "b", "c", "d", "e", "f", "g"];

  beforeEach(() => {
    let matches = tournamentMaker(teams);
  });

  it("returns an object containing the first leg of matches", () => {});
});

describe("support functions work correctly", () => {
  describe("isPowerOfTwo validates numbers correctly", () => {
    let testArr = [0, 1, 2, 3, 4, 7, 8, 9, 10, 256, -16, "a"];

    it("returns the expected answer", () => {
      expect(testArr.map(e => isPowerOfTwo(e))).toMatchSnapshot();
    });
  });
});
