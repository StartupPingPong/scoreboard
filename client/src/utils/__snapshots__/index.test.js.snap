// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`support functions work correctly isPowerOfTwo validates numbers correctly returns the expected answer 1`] = `
Array [
  false,
  false,
  true,
  false,
  true,
  false,
  true,
  false,
  false,
  true,
  false,
  false,
]
`;

exports[`support functions work correctly previousPowerOfTwo retuns the immediate power of two less than the input returns valid powers of two 1`] = `
Array [
  2,
  2,
  4,
  16,
  0,
  0,
]
`;
